pennylane_lightning/lightning_qubit.py

```diff
@@ -116,21 +116,20 @@ def apply(self, operations, rotations=None, **kwargs):
             super().apply(operations, rotations=rotations, **kwargs)
             return
 
-        for i, operation in enumerate(operations):  # State preparation is currently done in Python
+        # State preparation is currently done in Python
+        if isinstance(operations[0], QubitStateVector):
+            self._apply_state_vector(operations[0].parameters[0], operations[0].wires)
+            del operations[0]
+        elif isinstance(operations[0], BasisState):
+            self._apply_basis_state(operations[0].parameters[0], operations[0].wires)
+            del operations[0]
+
+        for operation in operations[1:]:
             if isinstance(operation, (QubitStateVector, BasisState)):
-                if i == 0:
-
-                    if isinstance(operation, QubitStateVector):
-                        self._apply_state_vector(operation.parameters[0], operation.wires)
-                    else:
-                        self._apply_basis_state(operation.parameters[0], operation.wires)
-
-                    del operations[0]
```

Comment:
Oof, yeah that's a bug...

Reply:
Hmm, it seems to have been okay as it was performed when `if isinstance(operation, (QubitStateVector, BasisState)) and i == 0` is `True`.